internal/app/app.go

```diff
@@ -129,25 +133,64 @@ func Run(ctx context.Context, osArgs []string, h *common.Health, optLoggerConfig
 
 	logger.Infoln("starting workers...")
 
-	h.AddReadiness()
-	eg.Go(func() error {
-		jp := journald.Processor{
-			BootID:    bootID,
-			MachineID: mid,
-			NodeName:  nodename,
-			Distro:    distro,
-			EventW:    eventWriter,
-			Logins:    logins,
-			CurrentTS: lastReadJournalTS,
-			Health:    h,
-		}
-
-		err := jp.Read(groupCtx)
-		if logger.Level().Enabled(zap.DebugLevel) {
-			logger.Debugf("journald worker exited (%v)", err)
-		}
-		return err
-	})
+	if distro == util.DistroRocky {
+		eg.Go(func() error {
+			// Create a tail
+			t, err := tail.TailFile(
+				"/var/log/secure", tail.Config{Follow: true, ReOpen: true})
+			if err != nil {
+				panic(err)
+			}
+			r := rocky.RockyProcessor{}
+
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case line := <-t.Lines:
+					logger.Infoln("calling process for rocky")
+					pm, err := r.Process(ctx, line.Text)
+					if err != nil {
+						logger.Errorf("error processing rocky secure logs %s", err.Error())
+						continue
+					}
+					if pm.PID != "" {
+						processors.ProcessEntry(&processors.ProcessEntryConfig{
```

Comment:
if there is a pid we call ProcessEntry only possible if the regex in `rockyprocessor` is matched.

Reply:
will do

Reply:
I see you added a log message. Looking at the other usage of `processors.ProcessEntry` in `journald`, it returns an error. I think you should stay consistent with that pattern.

Reply:
will do